### PR TITLE
Task: A tiny JS-World with OOP

### PR DESCRIPTION
**Pull request by KaterinaMykhailyk**
_Replace task name and links as appropriate and remove this line._

### A Tiny JS-World with OOP

[Demo](https://KaterinaMykhailyk.github.io/oop_exercise) |
[Code base](https://github.com/ThePickwickPapers/frontend-2019-p2p/blob/a-tiny-js-world-with-oop/submissions/katyamykhailik/oop_exercise/index.js)

The code is submitted in its own sub-directory and in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by Zakkarat** [comment]
Demo page is not working.

**Comment by KaterinaMykhailyk** [comment]
> @KaterinaMykhailyk Thank for your submission! Please, fix demo. Also you can delete `xml` files, they are doesn't necessary for this task.

Link was fixed. Thank you.

File: submissions/katyamykhailik/oop_exercise/index.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    constructor (name, gender, says, species) {
+        this.name = name;
+        this.gender = gender;
+        this.species = species;
+        this.says = says;
+        this.friend = "no friends yet :( "
+    }
+
+    toString(){
+        let inhabitantProfile = [this.name, this.species, this.gender, this.says, `friends: ${this.friend}`, ""];
```
Comment (o-msh) [inline submissions/katyamykhailik/oop_exercise/index.js]: You can use shortly instruction
```javascript
return [
 this.name,
 etc.
].join
```

Comment (o-msh) [reply submissions/katyamykhailik/oop_exercise/index.js]: Also in following methods.

File: submissions/katyamykhailik/oop_exercise/index.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+class CatWoman extends Human {
+    constructor(name, gender, says){
+        super(name, gender, says);
+        this.says = `${new Cat().says} & ${says}`;
+        this.species = "humananimal";
+    }
+}
+
+let catwoman = new CatWoman ("Irena", "female", "Hello!");
```
Comment (o-msh) [inline submissions/katyamykhailik/oop_exercise/index.js]: Use `camelCase` for defining variable.

File: submissions/katyamykhailik/oop_exercise/index.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+let cat = new Cat ("Tom", "male");
+let dog = new Dog ("Spike", "male");
+let boy = new Human ("Dmitriy", "male", "I am happy!");
+let girl = new Human ("Maria", "female", "I am here!");
+girl.makeFriendWith(cat);
+
+let inhabitants = Array.of(catwoman, cat, dog, boy, girl);
```
Comment (o-msh) [inline submissions/katyamykhailik/oop_exercise/index.js]:
```javascript
[catWoman, cat, etc].forEach
```

File: submissions/katyamykhailik/oop_exercise/index.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+inhabitants.forEach(inhabitant => print(inhabitant));
```
Comment (o-msh) [inline submissions/katyamykhailik/oop_exercise/index.js]: Please, add a new line. You can also configure your IDE.

File: submissions/katyamykhailik/oop_exercise/index.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        this.says = `${new Cat().says} & ${says}`;
+        this.species = "humananimal";
+    }
+}
+
+let catwoman = new CatWoman ("Irena", "female", "Hello!");
+let cat = new Cat ("Tom", "male");
+let dog = new Dog ("Spike", "male");
+let boy = new Human ("Dmitriy", "male", "I am happy!");
+let girl = new Human ("Maria", "female", "I am here!");
```
Comment (o-msh) [inline submissions/katyamykhailik/oop_exercise/index.js]: Each species must be represented with its own class. Please, declare `Man` and `Woman` classes.

**Pull request closed without merging**